RevenueCatUI/Helpers/Variables.swift

```diff
@@ -19,45 +20,50 @@ protocol VariableDataProvider {
 
 }
 
+struct VariableMatch {
+    let variable: String
+    let range: Range<String.Index>
+}
+
 /// Processes strings, replacing `{{variable}}` with their associated content.
+@available(iOS 16.0, macOS 13.0, tvOS 16.0, *)
 enum VariableHandler {
 
     static func processVariables(
         in string: String,
         with provider: VariableDataProvider
     ) -> String {
+        let matches = extractVariables(from: string)
         var replacedString = string
-        let range = NSRange(string.startIndex..., in: string)
-        let matches = Self.regex.matches(in: string, options: [], range: range)
-
-        for match in matches.reversed() {
-            let variableNameRange = match.range(at: 1)
-            if let variableNameRange = Range(variableNameRange, in: string) {
-                let variableName = String(string[variableNameRange])
-                let replacementValue = provider.value(for: variableName)
-
-                let adjustedRange = NSRange(
-                    location: variableNameRange.lowerBound.utf16Offset(in: string) - Self.pattern.count / 2,
-                    length: string.distance(from: variableNameRange.lowerBound,
-                                            to: variableNameRange.upperBound) + Self.pattern.count
-                )
-                let replacementRange = Range(adjustedRange, in: replacedString)!
 
-                replacedString = replacedString.replacingCharacters(in: replacementRange, with: replacementValue)
-            }
+        for variableMatch in matches.reversed() {
+            let replacementValue = provider.value(for: variableMatch.variable)
+            replacedString = replacedString.replacingCharacters(in: variableMatch.range, with: replacementValue)
         }
 
         return replacedString
     }
 
-    private static let pattern = "{{  }}"
-    // Fix-me: this can be implemented using the new Regex from Swift.
-    // This regex is known at compile time and tested:
-    // swiftlint:disable:next force_try
-    private static let regex = try! NSRegularExpression(pattern: "\\{\\{ (\\w+) \\}\\}", options: [])
+    private static func extractVariables(from expression: String) -> [VariableMatch] {
+        let variablePattern = Regex {
+            OneOrMore {
+                "{{ "
+                Capture {
+                    OneOrMore(.word)
+                }
+                " }}"
+            }
+        }
+
+        return expression.matches(of: variablePattern).map { match in
+            let (_, variable) = match.output
+            return VariableMatch(variable: String(variable), range: match.range)
```

Comment:
the neat thing here is that the range is for the full expression, the variable is the content of the `Capture`.